tests/document/indexes.py

```diff
@@ -911,22 +910,26 @@ class BlogPost(Document):
             meta = {'db_alias': 'test_indexes_after_database_drop'}
 
         try:
+            # Create Post #1
+            BlogPost(title='test1', slug='test').save()
+            # Re-create Post #1
+            with self.assertRaises(NotUniqueError):
+                BlogPost(title='test1', slug='test').save()
+
             BlogPost.drop_collection()
 
-            # Create Post #1
-            post1 = BlogPost(title='test1', slug='test')
-            post1.save()
+            # Create Post #1 again
+            BlogPost(title='test1', slug='test').save()
 
             # Drop the Database
             connection.drop_database('tempdatabase')
```

Comment:
Thanks for checking. I'll respond in the main thread given that it requires a higher-level discussion.